File: Movil/mobile/src/main/java/com/example/movil/models/AuthData.kt
package com.example.movil.models

class AuthData(
    val UsuarioId: Int,
    val Usuario: String,
    val PerfilId: Int,
    val TiempoSesion: Int,
    val TiempoInactividad: Int,
    val sesion: Sesion,
    val Bearer: String
) {
    constructor(data: Map<String, Any?>) : this(
        UsuarioId = (data["usuarioId"] as? Double)?.toInt() ?: 0,
        Usuario = data["usuario"] as? String ?: "",
        PerfilId = (data["perfilId"] as? Double)?.toInt() ?: 0,
        TiempoSesion = (data["tiempoSesion"] as? Double)?.toInt() ?: 0,
        TiempoInactividad = (data["tiempoInactividad"] as? Double)?.toInt() ?: 0,
        sesion = Sesion(data["sesion"]),
        Bearer = data["bearer"] as? String ?: ""
    )
}

File: Servicios/Security/Partial/AuthData.cs
using Servicios.Models.Tables;

namespace Servicios.Security.Partial
{
    public class AuthData
    {
        public int UsuarioId { get; set; }
        public string Usuario { get; set; }
        public int PerfilId { get; set; }
        public int TiempoSesion { get; set; }
        public int TiempoInactividad { get; set; }
        public Sesion sesion { get; set; }
        public string Bearer { get; set; }

        public AuthData(Usuario usuario)
        {
            this.UsuarioId = usuario.UsuarioId;
            this.Usuario = usuario.Cuenta;
        }

        public void SetSesion(Sesion sesion)
        {
            this.sesion = sesion;
        }
    }
}
